### PR TITLE
glamour-391 Improved import for Glamour Website 

### PR DESCRIPTION
**Pull request by rossralev**

### CODE REVIEW

File: src/Filter/IfThen.php [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        $output = $input;
+
+        foreach ($parameters as $param) {
+            $field = $param['field'] ?? false;
+
+            // Digs deeper in to the field or use the field value for the check
+            $value = Collection::get($output, $field, $output);
+            $state = true;
+
+            // Made like this to exten with the other Checks ne, gt, lt ...
```
Comment (ToBe998) [inline src/Filter/IfThen.php]: Using a fixed location for the condition type and a fixed location for it's evaluation parameter would make the code less convoluted (reduces need for all those issets). Also a switch/case would improve readability further.

File: src/Filter/ImportKrakenImageservice.php [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        foreach($variants as $variant){
+
+            if(!isset($variant["url"]) || !$variant["url"])
+                continue;
+
+            $diff = abs($target - $variant["aspectRatio"]);
+
+            if($bestDiff === false || $diff < $bestDiff){
+                $okImage = $variant;
+                $bestImage = $width <= $variant['width'] && $height <= $variant['height'] ? $variant : $bestImage;
```
Comment (ToBe998) [inline src/Filter/ImportKrakenImageservice.php:74]: move min height/width check outside diff check identical to the is-url-set check above.

File: src/Filter/Sections/TeaserSection.php [hunk 3]
```diff
@@ -0,0 +1,66 @@
+<?php
```
Comment (ToBe998) [inline src/Filter/Sections/TeaserSection.php:1]: Is cloning the whole importer here necessary? Only checking if the teaser is already present and link to that one might be sufficient. If duplication is necessary, at least make sure the ignore on duplicatze check is working as with the normal import.

File: src/Format/Krakenapi.php [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+    /**
+     * @param $filters
+     * @param int $limit
+     * @param int $offset
+     * @param array $order
+     * @return array|mixed
+     * @throws \CND\KrakenSDK\Exception
+     */
+    protected function requestKraken($filters, $limit = 20, $offset = 0, $order = []){
+        /*$cacheKey = md5(serialize([$filters,$limit,$offset,$order]));
```
Comment (ToBe998) [inline src/Format/Krakenapi.php]: uncomment or delete after finalization.

File: src/Service/Importer.php [hunk 5]
```diff
@@ -202,12 +209,11 @@ protected function extractValues($item, $channel, $config){
         return Collection::get($data, $config["source"], $config["default"]);
     }
 
-    protected function applyFilters($input, $config, $values, $item){
+    /*
```
Comment (ToBe998) [inline src/Service/Importer.php:212]: delete after finalization.

**Review by ToBe998 — CHANGES_REQUESTED** [review]
Requests for changes added...

File: src/Filter/ImportKrakenImageservice.php (hunk 2 above)
Comment (rossralev) [reply src/Filter/ImportKrakenImageservice.php:74]: We need this inside of the check, as the only possible image to fit the requirements may has a smaller size but still  has a perfect aspect ratio.

File: src/Filter/Sections/TeaserSection.php (hunk 3 above)
Comment (rossralev) [reply src/Filter/Sections/TeaserSection.php:1]: We need it as the teaser import needs to stay configureable, and has the same configurations as the normal importer.

Comment (ToBe998) [reply src/Filter/Sections/TeaserSection.php:1]: As discussed. We go the lazy route for internal Teasers with sugs.

**Pull request merged**